### PR TITLE
test(backend): add comprehensive unit and integration tests

### LINKED ISSUE #179: [QA][Backend] Uzupełnienie brakujących testów backendu
## Co
Dodanie brakujących testów backendu.

## Acceptance criteria
- [ ] Kluczowe komponenty backendu posiadają odpowiednie testy.
- [ ] Testy obejmują poprawne działanie oraz najważniejsze przypadki błędne.
- [ ] Wszystkie istniejące i dodane testy przechodzą poprawnie.
- [ ] Pokrycie testami zostało zweryfikowane i jest na uzgodnionym poziomie.

### PR DESCRIPTION
**Pull request by PR713**
## What & why

Adds broad backend test coverage (unit + integration with Testcontainers) for applications workflow, admin review, events, posts, profiles/tags, auth endpoints, Keycloak adapters, and security handlers. Closes gaps where only alumni/profile endpoints were tested.

## Scope

- [x] Backend (`backend/`)
- [ ] Web (`frontend/apps/web`)
- [ ] Mobile (`frontend/apps/mobile`)
- [ ] Shared packages / tooling / CI

## Acceptance criteria

- [x] Missing backend unit and integration tests added across controllers, services, domain logic, exception handling, and security
- [x] Tests cover happy paths, error/edge cases, and authorization where relevant
- [x] Existing tests preserved; full suite passes locally with Docker
- [x] No JaCoCo or other coverage reporting tooling added

## Tests

- [x] I added or updated automated tests for this change
- [x] Existing tests still pass locally
- [ ] No new tests

### CODE REVIEW

File: backend/src/test/java/pl/edu/agh/backend/security/handler/BffAuthenticationSuccessHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        when(authorizedClientService.loadAuthorizedClient("keycloak-mobile", auth.getName()))
+                .thenReturn(client);
+
+        handler.onAuthenticationSuccess(request, response, auth);
+
+        assertThat(response.getRedirectedUrl()).startsWith("myapp:///login#at=");
+        assertThat(response.getRedirectedUrl()).contains("rt=");
+        verify(authorizedClientService).removeAuthorizedClient("keycloak-mobile", auth.getName());
+        assertThat(SecurityContextHolder.getContext().getAuthentication()).isNull();
+    }
```
[the remark this answers is not in the document]
Comment (PR713) [reply backend/src/test/java/pl/edu/agh/backend/security/handler/BffAuthenticationSuccessHandlerTest.java]: done

**Review by Magmusacy — APPROVED** [review]
LGTM
